olive/cli/perf_tuning.py

```diff
@@ -0,0 +1,314 @@
+        )
+
+        sub_parser.add_argument(
+            "--output_path",
+            type=str,
+            default="perf_tuning_output",
+            help="Path to save the tuned inference settings.",
+        )
+        sub_parser.add_argument(
+            "--tempdir", default=None, type=str, help="Root directory for tempfile directories and files"
+        )
+        sub_parser.set_defaults(func=PerfTuningCommand)
+
+    @staticmethod
+    def perf_tuning_template():
+        with PERF_TUNING_TEMPLATE.open() as f:
+            return yaml.safe_load(f)
+
+    def _update_default_data_config_params(self, default_data_config) -> Dict:
+        data_config = deepcopy(default_data_config)
+        load_dataset_keys = (
+            "seq_len",
+            "past_seq_len",
+            "max_seq_len",
+            "shared_kv",
+            "generative",
+            "ort_past_key_name",
+            "ort_past_value_name",
+            "trust_remote_code",
+            "max_samples",
+        )
+        dataloader_keys = ("fields_no_batch", "batch_size")
+        args_dict = vars(self.args)
+        load_dataset_params = {k: args_dict[k] for k in load_dataset_keys if args_dict[k] is not None}
+        load_dataset_params["model_name"] = self.args.hf_model_name
+        dataloader_params = {k: args_dict[k] for k in dataloader_keys if args_dict[k] is not None}
+
+        data_config["load_dataset_config"] = load_dataset_params
+        data_config["dataloader_config"] = dataloader_params
+
+        # special field predict_with_kv_cache to choose the data container type
+        # from TransformersPromptDummyDataContainer and TransformersTokenDummyDataContainer
+        data_config["type"] = (
+            "TransformersTokenDummyDataContainer"
+            if self.args.predict_with_kv_cache
+            else "TransformersPromptDummyDataContainer"
+        )
+        return data_config
+
+    def _update_pass_config(self, default_pass_config) -> Dict:
+        pass_config = deepcopy(default_pass_config)
+        pass_config_keys = (
+            "cpu_cores",
+            "io_bind",
+            "enable_cuda_graph",
+            "execution_mode_list",
+            "opt_level_list",
+            "trt_fp16_enable",
+            "intra_thread_num_list",
+            "inter_thread_num_list",
+            "extra_session_config",
+        )
+        args_dict = vars(self.args)
+        pass_config.update({k: args_dict[k] for k in pass_config_keys if args_dict[k] is not None})
+        return pass_config
+
+    def _get_data_config(self, template_config=None) -> Dict:
+        if not self.args.data_config_path:
+            if template_config is None:
+                raise ValueError("Template config is required when data config is not provided.")
+            return self._update_default_data_config_params(template_config["data_configs"][0])
+        return DataConfig.parse_file_or_obj(self.args.data_config_path)
```

Comment:
Btw, might have to update the name of the data config here or the pass's param to match the user given name?